### DISCUSSION TITLE
ComfyUI compatibility when?

### DISCUSSION THREAD

**Discussion by clacheek** [DominikDoom/a1111-sd-webui-tagcomplete#246] [Ideas]
please make it compatible with ComfyUI

**Comment by DominikDoom** [comment]
Sadly, this is easier said than done. As discussed in #218, the extension currently isn't standalone and heavily depends on some aspects of the webui for both functional and display purposes. This starts at the folder structure it relies on, some code provided by the webui, the Gradio UI components I attach to, etc.

Most of that is not easily transferrable to ComfyUI, so to make it compatible with it would essentially require a full rewrite or a new project with separate code.
I'm currently quite busy with studying and work, so it would be quite a while before I could start working on it.
So it won't happen in the foreseeable future.

I would recommend using https://github.com/pythongosssss/ComfyUI-Custom-Scripts instead, it has an autocomplete feature built in.
It's not as feature-rich in comparison, but the autocompletion itself works well enough.

**Reply by clacheek** [reply]
Oh wow, well that's okay, take your time! And thanks for the recommendation, I will try it.
